addons/website_mail/static/src/js/website_mail.js

```diff
@@ -0,0 +1,193 @@
+
+    var qweb = core.qweb;
+    var _t = core._t;
+
+    /**
+     * Widget WebsiteMailThread
+     *
+     *  - Fetch message fron controller
+     *  - Display chatter: pager, total message, composer (according to access right)
+     *  - Provider API to filter displayed messages
+     */
+    var WebsiteMailThread = Widget.extend({
+        template: 'website_mail.chatter',
+        events: {
+            "click .o_website_mail_pager_btn": 'on_click_pager'
+        },
+
+        init: function(parent, options){
+            this._super.apply(this, arguments);
+            console.log($(parent).data());
+            this.options = _.defaults($(parent).data(), {
+                'allow_composer': true,
+                'display_composer': false,
+                'csrf_token': odoo.csrf_token,
+                'message_count': 0,
+                'pager_step': 10,
+                'pager_scope': 5,
+                'pager_start': 1,
+                'is_user_public': true,
+                'is_user_publisher': false,
+                'domain': [],
+            }, options || {});
+            console.log('options=', this.options);
+            this.set('messages', []);
+            this.set('message_count', this.options['message_count']);
+            this.set('current_page', this.options['pager_start']);
+            this.set('pager', {});
+            this.set('domain', this.options['domain']);
+        },
+        willStart: function(){
+            var self = this;
+            // load qweb template and init data
+            return $.when(
+                ajax.jsonRpc('/website_mail/init', 'call', {
+                    'res_model': this.options['res_model'],
+                    'res_id': this.options['res_id'],
+                    'limit': this.options['pager_step'],
+                    'domain': this.get('domain'),
+                }),
+                this._load_templates()
+            ).then(function(result){
+                // bind events
+                self.on("change:messages", self, self.render_messages);
+                self.on("change:message_count", self, function(){
+                    self.render_message_count();
+                    self.set('pager', self._pager(self.get('current_page')));
+                });
+                self.on("change:pager", self, self.render_pager);
+                self.on("change:current_page", self, self.on_change_current_page);
+                self.on("change:domain", self, self.on_change_domain);
+                // set options and parameters
+                self.options = _.extend(self.options, result['options'] || {});
+                self.set('message_count', self.options['message_count']);
+                self.set('messages', self.preprocess_messages(result['messages']));
+            });
+        },
+        _load_templates: function(){
+            return ajax.loadXML('/website_mail/static/src/xml/website_mail.xml', qweb);
+        },
+        fetch_message: function(domain){
```

Comment:
Shouldn't we use the same naming convention, and use message_fetch, message_fetch_prepare, ... ?